pvanalytics/quality/time.py

```diff
@@ -184,3 +184,123 @@ def shifts_ruptures(event_times, reference_times, period_min=2,
     # localize the shift_amount series to the timezone of the input
     shift_amount = shift_amount.tz_localize(event_times.index.tz)
     return shift_amount != 0, shift_amount
+
+
+def dst_dates(dates, tz):
+    """
+    Get the dates on which daylight savings transitions occur in
+    the timezone `tz`.
+
+    Parameters
+    ----------
+    dates : DatetimeIndex
+        A datetime index with one timestamp per day.
+    tz : str
+        A timezone string (e.g. 'America/Denver' or 'CET')
+
+    Returns
+    -------
+    Series
+        Boolean Series with True on days where daylight savins transitions
+        occur.
+    """
+    midnight = pd.Series(
+        pd.DatetimeIndex(dates.tz_localize(None).date),
+        index=dates
+    )
+    noon = midnight + pd.Timedelta(hours=12)
+    noon = noon.dt.tz_localize(tz)
+    dst_shift = noon.apply(lambda t: t.tzinfo.dst(t).total_seconds() / 3600)
+    return dst_shift.diff().fillna(0) != 0
+
+
+def _has_dst(events, date, window, min_difference):
+    """Return True if `events` appears to have a daylight savings shift
+    on `date`.
+
+    Parameters
+    ----------
+    events : Series
+        Series of timestamps, one per day.
+    date : Timestamp
+        Timestamp corresponding to midnight on the day to check for daylight
+        savings shifts.
+    window : int
+        Number of days before and after `date` to use when checking for
+        for shifts in `events`
+    min_different : int
+        Minimum difference between mean event time before `date` and mean
+        event time after `date` for a shift tom be detected.
+
+    Returns
+    -------
+    bool
+        Whether or not a shift of at least `min_difference` minutes was
+        detected in the the event times on `date`.
+    """
+    before = events[date - window:date - pd.Timedelta(days=1)]
+    after = events[date:date + window - pd.Timedelta(days=1)]
+    if len(before) == 0 or len(after) == 0:
+        raise ValueError(f"No data at {date}. "
+                         "Consider passing a larger `window`.")
+    before = before.dt.hour * 60 + before.dt.minute
+    after = after.dt.hour * 60 + after.dt.minute
+    return abs(before.mean() - after.mean()) > min_difference
+
+
+def has_dst(events, tz, window=7, min_difference=45):
+    """Return True if `events` appears to have daylight savings shifts
+    at the dates on which `tz` transitions to or from daylight savings
+    time.
+
+    Compares the mean event time in minutes since midnight over the
+    `window` days before and after each date in `shift_dates`. If the
+    difference is greater than `min_difference` then a shift has occurred
+    on that date.
+
+    Parameters
+    ----------
+    events : Series
+        Series with one timestamp for each day. The timestamp should
+        correspond to an event that occurs at roughly the same time on
+        each day, and shifts with daylight savings transitions. For example,
```

Comment:
```suggestion
        each day. For example,
```
I don't think it's required that the event timestamp changes with daylight savings, that would imply that `events` is localized to a daylight savings aware timezone.

Maybe add `events need not be localized`

Reply:
That's true. I think I was trying to communicate that the event whose time is being recorded should occur at roughly the same UTC time each day, and therefore be subject to daylight savings shifts when recorded in local time (i.e. you could pass a series with noon as the event, but it wouldn't do any good because that event shifts with DST).

`events need not be localized` plus the examples of what kind of events we expect is probably sufficient.